src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -802,6 +804,45 @@ class spell_mage_water_elemental_freeze : public SpellScript
     }
 };
 
+// 210824 Touch of the Magi (Aura) - SPELL_MAGE_TOUCH_OF_THE_MAGI_AURA
+class spell_mage_touch_of_the_magi_aura : public AuraScript
+{
+    PrepareAuraScript(spell_mage_touch_of_the_magi_aura);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_MAGE_TOUCH_OF_THE_MAGI_EXPLODE });
+    }
+
+    void HandleProc(AuraEffect* aurEff, ProcEventInfo& eventInfo)
+    {
+        DamageInfo* damageInfo = eventInfo.GetDamageInfo();
+        if (damageInfo)
+            if (damageInfo->GetAttacker() == GetCaster() && damageInfo->GetVictim() == GetTarget())
```

Comment:
codestyle: wrap in brackets (you do it on first required inner level, then all the way to the outermost one)